lib/drivers/medtronic600/NGPHistoryParser.js

```diff
@@ -315,20 +329,64 @@ class BloodGlucoseReadingEvent extends NGPHistoryEvent {
     return this.eventData.readUInt16BE(0x0C);
   }
 
-  // See NGPUtil.NGPConstants.BG_UNITS
   get bgUnits() {
     // bgValue is always in mg/dL. bgUnits tells us which units the device is set in.
     // eslint-disable-next-line no-bitwise
-    return this.eventData[0x0B] & 1 ? 0 : 1;
+    return this.eventData[0x0B] & 1 ?
+      NGPUtil.NGPConstants.BG_UNITS.MG_DL :
+      NGPUtil.NGPConstants.BG_UNITS.MMOL_L;
+  }
+
+  get calibrationFlag() {
+    // eslint-disable-next-line no-bitwise
+    return (this.eventData[0x0B] & 2) === 2;
   }
 
   get bgLinked() {
     return this.meterSerialNumber !== '';
   }
 
-  get calibrationFlag() {
+  get isCalibration() {
+    return (this.bgSource === NGPUtil.NGPConstants.BG_SOURCE.SENSOR_CAL || this.calibrationFlag);
+  }
+}
+
+class ClosedLoopBloodGlucoseReadingEvent extends NGPHistoryEvent {
+  get bgValue() {
+    // bgValue is always in mg/dL.
+    return this.eventData.readUInt16BE(0x0B);
+  }
+
+  get bgUnits() {
+    // bgValue is always in mg/dL. bgUnits tells us which units the device is set in.
     // eslint-disable-next-line no-bitwise
-    return (this.eventData[0x0B] & 2) === 2;
+    return this.eventData[0x16] & 1 ?
+      NGPUtil.NGPConstants.BG_UNITS.MG_DL :
+      NGPUtil.NGPConstants.BG_UNITS.MMOL_L;
+  }
+
+  get bgReadingOrigin() {
+    // eslint-disable-next-line no-bitwise
+    return this.eventData[0x16] & 2 ?
+      NGPUtil.NGPConstants.BG_ORIGIN.MANUALLY_ENTERED :
+      NGPUtil.NGPConstants.BG_ORIGIN.RECEIVED_FROM_RF;
+  }
+
+  // See NGPUtil.NGPConstants.BG_CONTEXT
+  get bgReadingContext() {
```

Comment:
Ah, good idea! I'll add that in.
Sorry, I got confused between "Data model payload" and "Medtronic protocol payload" ;)